lib/puppet/type/cisco_interface_ospf.rb

```diff
@@ -240,34 +246,33 @@ def name
   # and message_digest_password combination
 
   validate do
-    fail("area must be supplied when ensure=present") if
-      self[:ensure] == :present and self[:area].nil?
+    fail('area must be supplied when ensure=present') if
+      self[:ensure] == :present && self[:area].nil?
 
-    if (self[:message_digest_key_id].nil?) and
-         (!self[:message_digest_algorithm_type].nil? or
-          !self[:message_digest_encryption_type].nil? or
-          !self[:message_digest_password].nil?)
+    if (self[:message_digest_key_id].nil?) &&
+       (!self[:message_digest_algorithm_type].nil? ||
+        !self[:message_digest_encryption_type].nil? ||
+        !self[:message_digest_password].nil?)
       fail("If message_digest_key_id is not present in the manifest, \
             the following attributes must not be present - \
             message_digest_algorithm_type, \
             message_digest_encryption_type, message_digest_password" )
     end
 
-    if (!self[:message_digest_key_id].nil?) and
-         (self[:message_digest_algorithm_type].nil? or
-          self[:message_digest_encryption_type].nil? or
-          self[:message_digest_password].nil?)
+    if (!self[:message_digest_key_id].nil?) &&
+       (self[:message_digest_algorithm_type].nil? ||
+        self[:message_digest_encryption_type].nil? ||
+        self[:message_digest_password].nil?)
       fail("If message_digest_key_id is present in the manifest, \
-	    the following attributes must also be present - \
+      the following attributes must also be present - \
             message_digest_algorithm_type, \
             message_digest_encryption_type, message_digest_password")
```

Comment:
Both methods are valid, but the current implementation **is** incorrect.

    irb

    fail("If message_digest_key_id is present in the manifest, \
       the following attributes must also be present - \
             message_digest_algorithm_type, \
             message_digest_encryption_type, message_digest_password")

    RuntimeError: If message_digest_key_id is present in the manifest,       the following attributes must also be present -              message_digest_algorithm_type,              message_digest_encryption_type, message_digest_password
	from (irb):9
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/bin/irb:12:in `<main>'

The indentation of the escaped new lines are being counted as spaces in the string.  Your method will work and is cleaner than the current implementation, so I agree we should change it.